### ISSUE TITLE
MySQL queries that use the same parameter in two places

### ISSUE THREAD

**Issue by alecbz** [kyleconroy/sqlc#777]
A MySQL query like:

```sql
SELECT * FROM things WHERE foo = sqlc.arg(foo) OR old_foo = sqlc.arg(foo)
```

gets compiled into:

```go
const query = `SELECT * FROM things WHERE foo = ? OR old_foo = ?`

func (q *Queries) GetThingByFoo(ctx context.Context, foo string) (Thing, error) {
  row := q.db.QueryRowContext(ctx, query, foo)
  ...
}
```

which causes MySQL to complain because only one argument is supplied when it was expecting two.

---

It seems like we should at least error out during code generation in this case, but even better would be to do:

```go
row := q.db.QueryRowContext(ctx, query, foo, foo)
```

**Comment by mightyguava** [comment]
I remember explicitly solving for something very similar in the Postgres Kotlin codegen when numbered parameters are used. It was, https://github.com/kyleconroy/sqlc/pull/300/commits/0f5afd35cd06c71efd3a8ed482016e55ddcda422, done by tracking `paramRefs` by order of occurrence (allowing duplicates) instead of keeping them as a sorted set. This code still seems to work in Dolphin! Maybe something similar can be done to solve the problem with numbered parameters?

**Comment by kyleconroy** [comment]
Fixed in #1296
